chromium_src/chrome/browser/ui/tabs/tab_renderer_data.cc

```diff
@@ -44,5 +45,15 @@ TabRendererData TabRendererData::FromTabInModel(const TabStripModel* model,
       data.should_themify_favicon = false;
     }
   }
+
+  // Show which tabs are unloaded.
+  if (!data.should_show_discard_status) {
+    content::WebContents* const contents = model->GetWebContentsAt(index);
+    using resource_coordinator::TabLoadTracker;
+    const auto loaded_state = TabLoadTracker::Get()->GetLoadingState(contents);    
+    if (loaded_state == TabLoadTracker::LoadingState::UNLOADED) {
```

Comment:
It seems to me like `TabLoadTracker::LoadingState::UNLOADED` is a bit more all-encompassing than `discarded`, but I'm not positive. But maybe it would be safer to go with a similar approach to what's used upstream:

```
  data.should_show_discard_status =
      memory_saver::IsURLSupported(contents->GetURL()) && contents->WasDiscarded();
```

Reply:
Indeed it has to be all encompassing because if a session is restored it can include unloaded tabs that require `TabLoadTracker::LoadingState::UNLOADED`.

`memory_saver::IsURLSupported` is not so consistent because it excludes unloaded tabs having URLs such as `brave://settings`.

Such is Edge's behavior.

@bsclifton I just noticed your edit deleted this line from the test plan.

> Restore a session so that we'll have unloaded tabs that weren't discarded.

It's required because restored & unloaded tabs do not have discard state.

Reply:
@is-this-c whoops - ok I understand. I put this back 😄 Feel free to edit your original top post - our QA will use those steps when verifying the issue.